lxd/storage_lvm.go

```diff
@@ -1285,6 +1285,13 @@ func (s *storageLvm) doContainerMount(project, name string) (bool, error) {
 	ourMount := false
 	if !shared.IsMountPoint(containerMntPoint) {
 		mountFlags, mountOptions := lxdResolveMountoptions(s.getLvmMountOptions())
+		if snap && lvFsType == "xfs" {
+			idx := strings.Index(mountOptions, "nouuid")
```

Comment:
Oh right, we already do this somewhere else in the lvm code. I vaguely remember having troubles with this during snapshots or something.

Reply:
Yeah, we do on snapshotstart, that's where I took it from. The problem is that for backups we don't use the traditional snapshot mountpoints so needed to plumb it to the container mount function.

This code is a giant mess, hopefully @monstermunchkin's rework will make this much easier to deal with.

Short version is, when we create a new container from an image, we want to generate a new UUID, when we mount a container or custom storage volume, no fancy mountflag, when we mount anything else, we need nouuid.